### ISSUE TITLE
Incorrect town value when it is 20 instead of 10

### ISSUE THREAD

**Issue by perwestling** [tobymao/18xx#7503]
When using promotion tiles variant in 18Rhl I can see that tiles 949 and Essen does not show any revenue for the town on the tile.
Checking the code they do have "town=revenue:20,loc=4". Tile 950 has a similar definition, but instead revenue:10, and it does appear on the tile.
![image](https://user-images.githubusercontent.com/2631151/159134493-1c4af0a9-99c0-43e5-ae20-c3876b591b8d.png)

Have not checked if this problem exists in any other title, but as far as I know there is nothing special with 18Rhl in these cases.

**Comment by roseundy** [comment]
There is something wrong with how the towns are being rendered on 949, 950 and Essen. They should be rendered as rectangles and not circles. I'm guessing there may be a problem with how the paths are specified.

**Comment by roseundy** [comment]
Here is the definition for 949:
```
 '949' =>
          {
            'count' => 1,
            'color' => 'gray',
            'code' => 'city=revenue:60,slots:3,loc:center;town=revenue:20,loc:4;path=a:0,b:_0;path=a:1,b:_0;'\
                      'path=a:2,b:_0;path=a:3,b:_0;path=a:4,b:_0;path=a:5,b:_0;label=DU',
          },
```
Note that there aren't any paths leading to "_1", which is the town. Basically, the rendering code thinks the town is just an unconnected revenue location. I suspect that the code doesn't deal with a "floating" town with revenue very well.

To fix this, the path between the city and edge 4 needs to be broken up into two pieces with the town in the middle.

**Comment by perwestling** [comment]
@roseundy, thanks for the help. I have now pushed up a correction that fixes these three tiles.
![image](https://user-images.githubusercontent.com/2631151/168472894-247c91fb-fe55-4c31-8da8-2cac5ab09db7.png)

I also included a change of using label R instead of Ratingen in the Ratingen variant, as the Ratingen label looked pretty bad.
This are tile 1910 and 1911.

There are some small things that maybe should be fixed some day but these are general things.
![image](https://user-images.githubusercontent.com/2631151/168472950-4e7a862e-250b-45bc-a5b5-6954ea40bccd.png)
(I am thinking about R, Y and OO labels on hex G12, E12 and F13.)
